mempool/clist_mempool.go

```diff
@@ -362,14 +365,24 @@ func (mem *CListMempool) addTx(memTx *mempoolTx) {
 func (mem *CListMempool) removeTx(tx types.Tx, elem *clist.CElement, removeFromCache bool) {
 	mem.txs.Remove(elem)
 	elem.DetachPrev()
-	mem.txsMap.Delete(txKey(tx))
+	mem.txsMap.Delete(TxKey(tx))
 	atomic.AddInt64(&mem.txsBytes, int64(-len(tx)))
 
 	if removeFromCache {
 		mem.cache.Remove(tx)
 	}
 }
 
+// RemoveTxByKey removes a transaction from the mempool by its TxKey index.
+func (mem *CListMempool) RemoveTxByKey(txKey [TxKeySize]byte, removeFromCache bool) {
```

Comment:
Is there a need for exposing `removeFromCache`? Shouldn't it always be `true` here?

Reply:
If I don't understand wrong how cache and mempool works, I think exposing it makes sense. 

It could be that the caller wants to remove a TX from the mempool but keep it on cache in order to not receive it again. Does it makes sense to you @erikgrinaker ?

Reply:
Yeah, that makes sense. Looks like the cache is only used to determine whether a tx should be added again, so that should work as expected.